backend/src/ee/services/group/group-service.ts

```diff
@@ -88,6 +88,13 @@ export const groupServiceFactory = ({
     if (!hasRequiredPriviledges)
       throw new ForbiddenRequestError({ message: "Failed to create a more privileged group" });
 
+    const existingGroup = await groupDAL.findOne({ orgId: actorOrgId, name });
+    if (existingGroup) {
+      throw new BadRequestError({
+        message: `Failed to create group with name '${name}'. Group with the same name already exists`
+      });
+    }
+
     const group = await groupDAL.create({
       name,
       slug: slug || slugify(`${name}-${alphaNumericNanoId(4)}`),
```

Comment:
let's put this as well as the `create group` below into a transaction block so that the DB read request is always served by the primary node and not by replicas